developer_guides/dsp_core/index.rst

```diff
@@ -0,0 +1,81 @@
+.. _dsp_core:
+
+DSP Cores Management
+####################
+
+More and more audio DSPs with multiple cores are implemented in the current
+SoC(System on Chip), to fully ultilize the bandwidth of all cores, and do good
+power saving with power on/off cores dynamically, we need a generic DSP cores
+management with SOF stack.
+
+Context -- pipeline, component, and DSP core
+********************************************
+
+Driver pipeline and widget
+==========================
+
+A pipeline usually contains one scheduler widget (or named 'pipeline widget in
+the FW') + several other widgets, all these widgets are 'swidget's in the
+driver side.
+
+Firmware pipeline, component and DSP core
+=========================================
+
+In the firmware side, each pipeline has a dedicated pipeline task(related to
+pipeline_copy()), to be scheduled on a specified DSP core (set in topology via
+scheduler widget's core).
+
+Each component (normal driver widget) can have its own task also, it can be
+specified to run on a specified DSP core, this is not neccesary to be the same
+one with the pipeline one.
+
+
+How to manage the DSP cores in the driver?
+******************************************
+
```

Comment:
do we need a ref-count for the main core? We can do that for symmetry, but I'd only let it hit 0 when switching the DSP off for various suspend cases. So, the difference is, that all (secondary cores) are ref-counted for any audio-streaming, that they're needed for, whereas the main core is kept on until suspend. Maybe just increment its use-count by one as an "OS core."

And then begin by explaining that: that we want to power secondary cores on and off when using them for audio.